pkg/cmd/server/api/helpers.go

```diff
@@ -337,7 +337,7 @@ func GetInternalKubeClient(kubeConfigFile string, overrides *ClientConnectionOve
 
 // TODO: clients should be copied and instantiated from a common client config, tweaked, then
 // given to individual controllers and other infrastructure components.
-func GetExternalKubeClient(kubeConfigFile string, overrides *ClientConnectionOverrides) (kclientsetexternal.Interface, *restclient.Config, error) {
+func GetExternalKubeClient(kubeConfigFile string, overrides *ClientConnectionOverrides) (*kclientsetexternal.Clientset, *restclient.Config, error) {
```

Comment:
Since here I am reusing CreateScheduler instead of reinventing the wheel, I have sent a PR to address this https://github.com/kubernetes/kubernetes/pull/49379 .